babel.config.js

```diff
@@ -1,25 +1,25 @@
 module.exports = {
-  "sourceMaps": true,
-  "presets": [
-    ["@babel/preset-env", {
-      "targets": {
-        "browsers": [
-          "last 2 versions"
-        ]
-      }
-    }],
-    "@babel/preset-typescript",
-    "@babel/preset-flow",
-    "@babel/preset-react"
-  ],
-  "plugins": [
-    ["@babel/plugin-proposal-decorators", {"legacy": true}],
-    "@babel/plugin-proposal-export-default-from",
-    "@babel/plugin-proposal-numeric-separator",
-    "@babel/plugin-proposal-class-properties",
-    "@babel/plugin-proposal-object-rest-spread",
-    "@babel/plugin-transform-flow-comments",
-    "@babel/plugin-syntax-dynamic-import",
-    "@babel/plugin-transform-runtime"
-  ]
+    "sourceMaps": true,
+    "presets": [
+        ["@babel/preset-env", {
+            "targets": {
+                "browsers": [
+                    "last 2 Chrome versions", "last 2 Firefox versions", "last 2 Safari versions"
+                ]
+            },
+        }],
+        "@babel/preset-typescript",
+        "@babel/preset-flow",
+        "@babel/preset-react"
+    ],
+    "plugins": [
+        ["@babel/plugin-proposal-decorators", {"legacy": false, decoratorsBeforeExport: true}],
```

Comment:
presumably making them not legacy fixes the crash? do we understand why?

also it looks like the proposal is now at stage 2 with export before decorators, so I think we're on the losing side at this point. We might want to give in & use the other syntax before we add them to every component.

Reply:
They're constructed completely differently in the transpiled code. Decorators are indeed stage 2, but also doing it not-legacy means we're closer to the TypeScript version which can work more easily.

The syntax is the same for usage, just different for definition. Though again, it's now closer to TypeScript's version.